week2/assign4/index.js

```diff
@@ -0,0 +1,219 @@
+const INIT_BALANCE =0;
+let HISTORY_LIST=[
+  {id:0,type:"income",category:"과외비",content:"10월 월급",amount:50000},
+  {id:1,type:"expense",category:"식비",content:"마라탕",amount:10000},
+  {id:2,type:"income",category:"과외비",content:"10월 월급",amount:50000},
+  {id:3,type:"expense",category:"식비",content:"커피",amount:6000}];
+
+  //list 추가를 위한 변수들
+let itemContainer;
+let type='income';
+let category;
+let content;
+let amount;
+let nowid=3;
+// let button;
+
+let totalIncome=0;
+let totalExpense=0;
+
+const listcontent = document.getElementById('lists'); //내역리스트 담는 ul태그
+const BALANCE = document.getElementById('balance');
+let balance=INIT_BALANCE;
+BALANCE.textContent= balance;
+
+//각 list의 정보 보여주는 함수 (중복해서 쓰여서 따로 뺌)
+const showList =(list)=>{
+  itemContainer = document.createElement('li');
+      
+  category = document.createElement('span');
+  category.textContent=list.category;
+  itemContainer.appendChild(category);
+
+  content = document.createElement('span');
+  content.textContent=list.content;
+  itemContainer.appendChild(content);
+
+  amount = document.createElement('span');
+  amount.textContent=list.amount;
+  itemContainer.appendChild(amount);
+  amount.classList.add(list.type);
+
+  const button = document.createElement('button');
+  button.textContent = 'x';
+  button.classList.add("xbtn");
+  itemContainer.appendChild(button);
+  //x버튼 클릭시 이벤트
+  button.addEventListener('click',()=>{
+    console.log(list.id,"삭제 버튼"); 
+    deleteItem(list.id);
+    console.log(HISTORY_LIST)
+  });
+
+  listcontent.appendChild(itemContainer);
+
+  
+}
+
+//총수익, 총지출, 잔액 구하기
+const calculateTotal=()=>{
+  totalExpense=0;
+  totalIncome=0;
+  HISTORY_LIST.forEach((item)=>{
+    //총수익, 총지출 구하기
+    if(item.type==="income"){
+      totalIncome += item.amount;
+    } else if(item.type === "expense"){
+      totalExpense +=item.amount;
+    }
+  })
```

Comment:
이 부분은 삼항 연산자로 표현해줄수도 있겠다!
```suggestion
item.type==="income" ? totalIncome += item.amount : totalExpense += item.amount
```
이런식으로?